rust/libnewsboat-ffi/src/matcher.rs

```diff
@@ -0,0 +1,86 @@
+use libc::c_char;
+use libc::c_void;
+use std::ffi::CString;
+use std::ffi::CStr;
+
+use abort_on_panic;
+use libnewsboat::matcher::{Matchable, Matcher};
+
+extern "C" {
+#[no_mangle]
+fn matchable_has_attribute(item: *const c_void, attr: *const c_char) -> bool;
+#[no_mangle]
+fn matchable_get_attribute(item: *const c_void, attr: *const c_char) -> *mut c_char;
+}
+
+struct ForeignMatchable {
+    ptr: *const c_void
+}
+
+impl Matchable for ForeignMatchable {
```

Comment:
Why not use `attribute(&self, attr: &str) -> Option<String>` to replace both `has_attribute` and `get_attribute`?